tests/func/test_tree.py

```diff
@@ -115,7 +115,7 @@ def test(self):
             [
                 (
                     self._root_dir,
-                    ["data_dir"],
+                    ["lib", "data_dir"],
```

Comment:
You are modifying unrelated tests, please see the comment above 🙂

Reply:
Disregarding the comment above; the changes were made because the addition of the `lib` directory changes the filesystem tree. If you'd like me to avoid modifying other tests I'd need to create a separate fixture which extends `TestDirFixture` and adds regular files.

Reply:
I guess there is a misunderstading here. I'm simply trying to say that the correct way to approach this is to modify exiting `erepo` fixture  or build on top of it (you can even avoid using var names like FOO,BAR and use plain string filename instead, no problem) to add files/dirs that you need.

Reply:
So let's clarify things to avoid confusion. How exactly do you want a basic smoke test setup to look?

Here are my assumptions:

- We create a test directory into which we want to use `dvc get` to retrieve and store files. The test directory should probably exist in `/tmp/`. Let's call this directory A.
- We want a remote git project to `dvc get` files from. Ideally this should be a local temporary directory as well to avoid going over the network for reproducibility and performance reasons. Let's call this directory B.
- We want to retrieve a file called `file.txt` from directory B.
- We use `dvc get` and afterwards we want to  assert that the filename and contents of the newly created file in directory A is equal to the file in directory B.

```
         Dir A                  Dir B


     +----------+            +----------+
     |          |            |          |
     |          |            |          |
1.   |          +<-----------+ file.txt |
     |          |            |          |
     +----------+            +----------+



     +----------+            +----------+
     |          |            |          |
2.   |          |            |          |
     |  file.txt|            |  file.txt|
     |          |            |          |
     +----------+            +----------+


3.     assert A/file.txt == B/file.txt
```

The questions that remain for me before implementation:

- is it OK to use a temporary tmp directory without any git or dvc setup for directory A, from which we `dvc get` files?
- can I use the erepo fixture to create directory B and fetch the file `code.py` from directory B. `code.py` doesn't seem to be checked into dvc (https://github.com/iterative/dvc/pull/2837#discussion_r349888881, https://github.com/iterative/dvc/blob/master/tests/conftest.py#L141-L174)

e.g

```py
import os
import tempfile

def test_get_regular_file(erepo):
    dir = tempfile.TemporaryDirectory()
    dst = os.path.join(dir.name, "code_imported")
    os.chdir(dir.name)
    Repo.get(erepo.root_dir, erepo.CODE, dst)
    assert filecmp.cmp(src, dst, shallow=False)
```

This would still require creating a nested directory in the `erepo fixture` in order to test `dvc get` with directories. Do you want me to create the nested directory or to create a new fixture specifically for these two tests?

Reply:
There is definitely a misunderstanding here, your current tests `test_get_regular_file `and `test_get_regular_dir` are totally fine, except for the fact that you are modifying TestDirFixture for no good reason. Just stop modifying it and either create those files in place as @Suor mentioned or modify erepo(or create a fixture based on it) to have those files. I'm not against the latter case, because we will soon have to test the same stuff for `dvc import` and we'll need the same files anyway, but we could do that when we actually need it.

Just to absolutely clarify this, here is an UNTESTED example of how an acceptable `test_get_regular_file` could look like:
```
def test_get_regular_file(repo_dir, erepo):
    src = "file"
    dst = src + "_imported"

    with open(os.path.join(erepo.root_dir, src), "w+") as fobj:
        fobj.write("something")

    erepo.scm.add([src])
    erepo.scm.commit("add file")

    Repo.get(erepo.root_dir, src, dst)

    assert os.path.exists(dst)
    assert os.path.isfile(dst)
    assert filecmp.cmp(src, dst, shallow=False)
```

Reply:
You can use:
```python
erepo.create(src, "something")
```
instead of `with open...` I guess.

Reply:
@Suor Right, forgot that erepo is not a Repo() instance. Thanks!